mountinfo/mounted_linux.go

```diff
@@ -40,8 +64,8 @@ func mounted(path string) (bool, error) {
 		return false, err
 	}
 	// Try a fast path, using openat2() with RESOLVE_NO_XDEV.
-	mounted, err := mountedByOpenat2(path)
-	if err == nil {
+	mounted, sure, err := mountedByOpenat2(path)
+	if mounted && sure && err == nil {
 		return mounted, nil
 	}
 	// Another fast path: compare st.st_dev fields.
```

Comment:
1. mountedByStat could be called from MountedFast
2. Moreover, how about calling MountedFast from Mounted?

Reply:
1. Sure! Let me make that change.
2. We will be calling `normalizePath` twice for every `mounted` call. We can avoid this by relying on Os build tags and have `Mounted` for different OS. would you like me to make that refactor as well?

Reply:
Ah sorry, I meant, Linux-specific `mounted` could call `MountedFast`. Regarding normalizePath, I honestly okay to call that multiple times. It wouldn't be that slow. Would it be?

Reply:
`mounted` from `mountinfo/mounted_linux.go` already have all this, What's needed is to remove the fallback, instead returning `sure == false`.